internal/pkg/term/progress/deploy.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Just a thought experiment - is this implementation specific to CloudFormation? If so, should we call this file CloudFormation?

Reply:
The idea for the `deploy` package was for it to be be agnostic of the underlying tool used to do the deployment (cfn vs. something else)

That's why I named this file `deploy.go`.  I agree that the `deploy.Resource` struct is a bit too CFN specific with `LogicalName` and `Type` as fields :| 
I thought that a resource name and  type would be there no matter what type of deployment